### PR TITLE
Sync only when CLOUDANT env enabled

### LINKED ISSUE #48: DB attempts to sync when CLOUDANT env isn't set
Cloudant sync shouldn't run when CLOUDANT env isn't set.

> [Mon Aug 22 2016 14:00:54 GMT-0400 (EDT)] - error: nlcDb.js: Error during sync of NLC training data with Cloudant. Will retry in 60 seconds. Error: getaddrinfo ENOTFOUND undefined undefined:443
>   at errnoException (dns.js:27:10)
>   at GetAddrInfoReqWrap.onlookup [as oncomplete](dns.js:78:26)

I don’t have any CLOUDANT environment variables set in my local Cloudbot, so it seems like this is running when the env isn’t set, thus the `undefined undefined:443` above.

### PR DESCRIPTION
**Pull request by jlpadilla**
Resolves #48

### CODE REVIEW

File: src/lib/nlcDb.js [hunk 1]
```diff
@@ -27,43 +27,44 @@ const pouch = new Promise((resolve, reject) => {
...
-						return doc.storageType !== 'private';
-					}
-				})
-				// .on('change', function(change){
-				// 	update = true;
-				// })
-				.on('complete', function(info){
-					logger.info(`${TAG}: Completed sync of NLC training data with Cloudant.`);
-					logger.debug(`${TAG}: Cloudant sync results.`, info);
+			if (!env.text && env.cloudantEndpoint && env.cloudantPassword) {
```
Comment (chambrid) [inline src/lib/nlcDb.js]: should this be `env.test`  --> https://github.com/ibm-cloud-solutions/hubot-ibmcloud-cognitive-lib/blob/master/src/lib/env.js#L28

Comment (jlpadilla) [inline src/lib/nlcDb.js]: Yes, typo.  Corrected it.

**Comment by jlpadilla** [comment]
Corrected typo. Also added a log message when Cloudant sync isn't enabled.

**Comment by jlpadilla** [comment]
Just noticed sync also requires env.cloudantKey. Adding it.

**Pull request merged**